src/encord_active/lib/project/project_file_structure.py

```diff
@@ -101,30 +173,22 @@ def label_row_structure(self, label_hash: str) -> LabelRowStructure:
         return LabelRowStructure(path=path, mappings=self._mappings)
 
     def data_units(
-        self, pattern: Optional[DataUnitLike] = None, include_label_row: bool = False
-    ) -> Iterator[models.DataUnit]:
+        self, where: Optional[DataUnitWhereInput] = None, include_label_row: bool = False
+    ) -> List[models.DataUnit]:
         PrismaConnection.set_project_file_structure(self)
-        to_include = {"label_row": True} if include_label_row else {}
+        to_include = DataUnitInclude(label_row=True) if include_label_row else None
         with PrismaConnection() as conn:
-            # add backwards compatibility code. Remove when Encord Active version is >= 0.1.60.
-            fill_missing_tables()
-            # end backwards compatibility code.
-            if pattern is None:
-                return iter(conn.dataunit.find_many(include=to_include))
+            _fill_missing_tables(self)
+            if where is None:
```

Comment:
[nitpick 💅 ]
`find_many()`'s `where` param is `Optional` too, so we can skip the if/else and always send the `where` value to `find_many()`.